projects/igniteui-angular/src/lib/grid/api.service.ts

```diff
@@ -136,6 +136,11 @@ export class IgxGridAPIService {
         }
     }
 
+    public get_row_index_in_data(id: string, rowID: any): number {
+        const grid = this.get(id) as IgxGridComponent;
+        return grid.primaryKey ? grid.data.map((record) => record[grid.primaryKey]).indexOf(rowID) : grid.data.indexOf(rowID);
+    }
+
```

Comment:
add check for the grid - if `this.get(id)` returns undefined next row will throw an error